drake_ros_introspection/include/drake_ros_introspection/simulator_monitor.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include <drake/systems/framework/context.h>
+#include <drake/systems/framework/event_status.h>
+#include <drake_ros_introspection/probes/probe_interface.h>
+
+namespace drake_ros_introspection {
+
+template <typename T>
+class SimulatorMonitor {
+ public:
+  explicit SimulatorMonitor(
+      std::vector<std::unique_ptr<probes::ProbeInterface<T>>> probes)
+      : probes_(std::move(probes)) {}
+
+  void Configure(const std::shared_ptr<rclcpp::Node>& node) {
+    for (auto& probe : probes_) {
+      probe->Configure(node);
+    }
+  }
+
+  operator std::function<
+      drake::systems::EventStatus(const drake::systems::Context<T>&)>() {
+    return [this](const drake::systems::Context<T>& context) {
+      auto status = drake::systems::EventStatus::DidNothing();
+      for (auto& probe : probes_) {
+        status.KeepMoreSevere(probe->Probe(context));
```

Comment:
> If someone has written a pure-Drake "monitor" then it cannot be used without being wrapped in a Probe subclass.

That's true if the user wants to include it within the `SimulatorMonitor` (and delegate event status aggregation), but nothing prevents them from having e.g. a lambda that invokes both the `SimulatorMonitor` and their own functional.

> I think it would be a little simpler as a pure-drake utility for combining multiple `Simulator` "monitor"s

Yeah, I see the value in a utility that helps combine multiple monitor functions.

> Instead of having a list of probes, it would have a list of `std::function< EventStatus(const Context< T > &)>`. Then pure-Drake "monitors" whose job was only to determine when to terminate the simulation could be used as well as `Probe` subsclasses that inspect the system.

Something has to own `Probe` instances though. Thinking out loud, we could go fully functional, sorting out `std::function` copyability requirement via `std::shared_ptr` instances. It was an arbitrary choice to do unique ownership and dynamic polymorphism over shared ownership and type erasure (though I will say that I tend to prefer consistent type systems).